onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -2339,6 +2339,44 @@ void RegisterContribSchemas() {
         updateOutputShape(ctx, 0, output_shape);
       });
 
+  static const char* DisentangledAttention_TRT_ver1_doc =
+      R"DOC(Disentangled Attention TensorRT Plugin.)DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(DisentangledAttentionPlugin)
+      .SetDomain(kOnnxDomain)
+      .SinceVersion(1)
+      .SetDoc(DisentangledAttention_TRT_ver1_doc)
+      .Input(0, "c2c_attention", "content-to-content attention tensor, QcKc^T.", "T") 
+      .Input(1, "c2p_attention", "content-to-position attention tensor, QcKr^T.", "T")
+      .Input(2, "p2c_attention", "position-to-content attention tensor, KcQr^T.", "T")
+      .Output(0, "disentangled_attention", "The disentangled attention output tensor.", "T")
+      .TypeConstraint("T", {"tensor(float)", "tensor(float16)"}, "Constrain input and output types to float tensors.")
+      .Attr("span", "Maximum relative distance, k.", AttributeProto::INT)
+      .Attr("factor", "Scaling factor applied to attention values, 1/sqrt(3d). d is hidden size per head = H/N. H is hidden size, N is number of heads.", AttributeProto::FLOAT)
+      .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+        // Type inference
+        using namespace ONNX_NAMESPACE;
+        propagateElemTypeFromInputToOutput(ctx, 0, 0); 
+
+        // Shape Inference
+        if (!hasInputShape(ctx, 0)) {
+          return;
+        }
+
+        auto& input0_shape = getInputShape(ctx, 0);
+        auto& input0_dims = input0_shape.dim();
+        if (input0_dims.size() != 3) {
+          fail_shape_inference("Input 0 shall be 3 dimensions");
+        }
+
+        // output dims is same as input[0] dims, i.e., regular c2c attention dims
```

Comment:
Below code can be replaced by ONNX_NAMESPACE::propagateShapeFromInputToOutput(ctx, 0, 0).

Reply:
Changed as suggested